asserts/fetcher.go

```diff
@@ -60,6 +63,34 @@ func NewFetcher(trustedDB RODatabase, retrieve func(*Ref) (Assertion, error), sa
 	}
 }
 
+type SeqFetcher interface {
+	// FetchSequence retrieves the assertion as indicated by its sequence reference.
+	FetchSequence(*AtSequence) error
+}
+
+// NewSeqFetcher creates a Fetcher which will use trustedDB to determine trusted
+// sequence forming assertions, and fetch assertions using retrieve, and then will pass
+// them to save. This is the sequence-forming variant of NewFetcher
+func NewSeqFetcher(trustedDB RODatabase, retrieve func(*Ref) (Assertion, error), retrieveSeq func(*AtSequence) (Assertion, error), save func(Assertion) error) SeqFetcher {
```

Comment:
NewSequenceFetcher

Reply:
I took the liberty to name it `NewSequenceFormingFetcher`